network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt

```diff
@@ -44,43 +45,54 @@ class RealWgServerApi @Inject constructor(
     @UnprotectedVpnControllerService private val wgVpnControllerService: WgVpnControllerService,
     private val serverDebugProvider: WgServerDebugProvider,
     private val netNetpEgressServersProvider: NetpEgressServersProvider,
+    private val netpRepository: NetworkProtectionRepository,
 ) : WgServerApi {
 
     override suspend fun registerPublicKey(publicKey: String): WgServerData? {
-        // This bit of code gets all possible egress servers which should be order by proximity, caches them for internal builds and then
-        // returns the closest one or null if list is empty
-        val selectedServer = wgVpnControllerService.getServers().map { it.server }
-            .also { fetchedServers ->
-                logcat { "Fetched servers ${fetchedServers.map { it.name }}" }
-                serverDebugProvider.cacheServers(fetchedServers)
-            }
-            .map { it.name }
-            .firstOrNull { serverName ->
-                serverDebugProvider.getSelectedServerName()?.let { userSelectedServer ->
-                    serverName == userSelectedServer
-                } ?: false
-            }
-
-        val userPreferredLocation = netNetpEgressServersProvider.updateServerLocationsAndReturnPreferred()
-        val registerKeyBody = if (selectedServer != null) {
-            RegisterKeyBody(publicKey = publicKey, server = selectedServer)
-        } else if (userPreferredLocation != null) {
-            if (userPreferredLocation.cityName != null) {
-                RegisterKeyBody(publicKey = publicKey, country = userPreferredLocation.countryCode, city = userPreferredLocation.cityName)
+        val registerKeyBody = try {
```

Comment:
This is one way of doing it. Another way would be to handle every 403 from every endpoint being called here (servers and locations). I figured this is the simpler approach